### DISCUSSION TITLE
Prisma with a monorepo setup with a different `DATABASE_URL` per package

### DISCUSSION THREAD

**Discussion by wmtrinu** [prisma/prisma#23757] [Q&A]
### Question

I currently have a **postgres** database that uses _row level security_. This database is used by two services that are built with NestJS.

- auth service
- core service

In order to interact correctly with the database and RLS, I need to use these two services with a different database account so that it can bypass RLS. (auth service needs to bypass)

Here is where the monorepo setup comes to play, I want to implement a monorepo with 4 packages:

- db (prisma)
- auth service (nestjs)
- core service (nestjs)
- web (next js app)

The idea is that both `auth` and `core` can connect to the same database using a different username, I was thinking on passing a different `DATABASE_URL` when starting the service.

Has anyone done this approach or can share how to do so? I did not find anything on the documentation and would like a but of guidance please

Thanks

**Comment by ludralph** [comment]
Hi @wmtrinu 👋 

Thank you for raising this question.

For passing a different `DATABASE_URL` when starting the service, you could potentially use environment variables to achieve this. You would set a different `DATABASE_URL` for each service in their respective environment files.

Here's a rough idea of how you could structure your monorepo:
```
/my-monorepo
  /packages
    /db
      /prisma
        schema.prisma
    /auth-service
      .env // DATABASE_URL for auth service
      /src
    /core-service
      .env // DATABASE_URL for core service
      /src
    /web
      /src
 ```

In your `schema.prisma` file, you would reference the `DATABASE_URL` like so:
```
datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}
```
Then, when starting each service, you would use its respective .env file which contains the correct `DATABASE_URL`.

Do try this out and let us know if it works for you.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
